test/parallel/test-http2-methods.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+const server = h2.createServer();
+
+const methods = [undefined, 'GET', 'POST', 'PATCH', 'FOO', 'A B C'];
```

Comment:
yep, the code does not currently do any validation, which is intentional for the time being. We'll need to go back in and add the necessary validation but we'll need to experiment with to find a method that doesn't kill performance.